### ISSUE TITLE
Ansible broken due to cffi package mismatch

### ISSUE THREAD

**Issue by jakubgs** [NixOS/nixpkgs#138715]
### Describe the bug
Ansible `2.11` is broken in `nixos-unstable` due to `cffi` package version mismatch:
```
 > ansible --version
ERROR! Unexpected Exception, this is probably a bug: Version mismatch: this is the 'cffi' package version 1.14.5, located in '/etc/profiles/per-user/jakubgs/lib/python3.8/site-packages/cffi/api.py'.  When we import the top-level '_cffi_backend' extension module, we get version 1.14.6, located in '/nix/store/lxigrwzrkz3wk2j06fakslf6zavwddhc-python3.9-cffi-1.14.6/lib/python3.9/site-packages/_cffi_backend.cpython-39-x86_64-linux-gnu.so'.  The two versions should be equal; check your installation.
the full traceback was:

Traceback (most recent call last):
  File "/nix/store/imv5hvnvjqkkf06dfa5drim4znhhvj9d-python3.9-ansible-core-2.11.4/bin/.ansible-wrapped", line 105, in <module>
    mycli = getattr(__import__("ansible.cli.%s" % sub, fromlist=[myclass]), myclass)
  File "/etc/profiles/per-user/jakubgs/lib/python3.9/site-packages/ansible/cli/__init__.py", line 21, in <module>
    from ansible.inventory.manager import InventoryManager
  File "/etc/profiles/per-user/jakubgs/lib/python3.9/site-packages/ansible/inventory/manager.py", line 38, in <module>
    from ansible.plugins.loader import inventory_loader
  File "/etc/profiles/per-user/jakubgs/lib/python3.9/site-packages/ansible/plugins/loader.py", line 23, in <module>
    from ansible.parsing.utils.yaml import from_yaml
  File "/etc/profiles/per-user/jakubgs/lib/python3.9/site-packages/ansible/parsing/utils/yaml.py", line 17, in <module>
    from ansible.parsing.yaml.loader import AnsibleLoader
  File "/etc/profiles/per-user/jakubgs/lib/python3.9/site-packages/ansible/parsing/yaml/loader.py", line 30, in <module>
    from ansible.parsing.yaml.constructor import AnsibleConstructor
  File "/etc/profiles/per-user/jakubgs/lib/python3.9/site-packages/ansible/parsing/yaml/constructor.py", line 28, in <module>
    from ansible.parsing.vault import VaultLib
  File "/etc/profiles/per-user/jakubgs/lib/python3.9/site-packages/ansible/parsing/vault/__init__.py", line 58, in <module>
    from Crypto.Cipher import AES as AES_pycrypto
  File "/nix/store/1s26g5972qi8ci87w8sp5fwmp5c94q28-python3.9-pycryptodome-3.10.1/lib/python3.9/site-packages/Crypto/Cipher/__init__.py", line 27, in <module>
    from Crypto.Cipher._mode_ecb import _create_ecb_cipher
  File "/nix/store/1s26g5972qi8ci87w8sp5fwmp5c94q28-python3.9-pycryptodome-3.10.1/lib/python3.9/site-packages/Crypto/Cipher/_mode_ecb.py", line 29, in <module>
    from Crypto.Util._raw_api import (load_pycryptodome_raw_lib,
  File "/nix/store/1s26g5972qi8ci87w8sp5fwmp5c94q28-python3.9-pycryptodome-3.10.1/lib/python3.9/site-packages/Crypto/Util/_raw_api.py", line 80, in <module>
    ffi = FFI()
  File "/etc/profiles/per-user/jakubgs/lib/python3.8/site-packages/cffi/api.py", line 54, in __init__
    raise Exception("Version mismatch: this is the 'cffi' package version %s, located in %r.  When we import the top-level '_cffi_backend' extension module, we get version %s, located in %r.  The two versions should be equal; check your installation." % (
Exception: Version mismatch: this is the 'cffi' package version 1.14.5, located in '/etc/profiles/per-user/jakubgs/lib/python3.8/site-packages/cffi/api.py'.  When we import the top-level '_cffi_backend' extension module, we get version 1.14.6, located in '/nix/store/lxigrwzrkz3wk2j06fakslf6zavwddhc-python3.9-cffi-1.14.6/lib/python3.9/site-packages/_cffi_backend.cpython-39-x86_64-linux-gnu.so'.  The two versions should be equal; check your installation.
```

### Steps To Reproduce

1. Use `ansible_2_11` from `nixos-unstable` after https://github.com/NixOS/nixpkgs/commit/eefb7248f40
2. See the error.

### Additional context
I'm not entirely sure if https://github.com/NixOS/nixpkgs/commit/eefb7248f40 is the cause of this, but it seems plausible.

When I use the commit right before it(https://github.com/NixOS/nixpkgs/commit/ca6cb24b966) everything works fine, but I went a bit back before all the other Python package updates and changes. I tried https://github.com/NixOS/nixpkgs/commit/4cc9b841 and it's already broken.

### Notify maintainers
@mweinelt

### Metadata
```console
 > nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.10.66, NixOS, 21.05.3216.ebf419e737b (Okapi)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.15`
 - channels(root): `"nixos-21.05.3216.ebf419e737b, nixos-unstable-21.11pre314199.f6e9e908ccb"`
 - channels(sochan): `"nixos-21.05.2521.7bbca9877ca, nixos-20.09-20.09.4393.068984c00e0, nixos-unstable-21.11pre309670.253aecf69ed"`
 - channels(jakubgs): `"nixos-21.05.3216.ebf419e737b, nixos-20.09-20.09.4393.068984c00e0, nixos-unstable-21.11pre316684.79c444b5bde"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module:
  - ansible_2_11
```

**Comment by mweinelt** [comment]
I backported ansible_2_11 in #138355, please use it directly from the release branch instead of picking it from unstable. I'll try to keep it updated consistently.

Other than that I can't reproduce your issue.

**Comment by jakubgs** [comment]
Okay, thanks, that's great. But I always wondered, what's the difference between `release-21.05` and `nixos-21.05`?

**Comment by mweinelt** [comment]
The `release-21.05` can be compared to `master`, it is a working branch, that is then built by hydra. If the jobset meets certain conditions that results in a channel bump. When that happens the `nixos-21.05` and `nixos-21.05-small` branches are updated. The `nixos-*` branches therefore reflect the state of the respective channels.

**Comment by jakubgs** [comment]
Okay, then I'll wait till it gets into `nixos-21.05` and test it out, the hopefully close this issue.

**Comment by W1M0R** [comment]
For anyone else experience this (or similar cffi mismatch issues), the solution for me was to ensure that all python packages and applications imported from nix, should come from the same nix channel.

I experienced this issue when I sourced `pgcli` (a python application) from a stable channel and `ansible`  from the unstable channel within the same `shell.nix` file. They introduced different python environments into the nix shell, causing the mismatch to occur. When I changed `pgcli` to also come from the same channel as `ansible`, the issue was resolved.
